en/lessons/visualizing-with-bokeh.md

```diff
@@ -0,0 +1,779 @@
+
+```python
+from bokeh.models import BoxAnnotation
+```
+
+To create the box, we first need to determine its coordinates. Coordinates for Bokeh annotations can be either absolute (i.e. positioned using screen units), meaning they always stay in one place, or they can be positioned in relation to data. Our annotations will all be positioned using data coordinates. 
+
+```python
+box_left = pd.to_datetime('6-6-1944')
+box_right = pd.to_datetime('16-12-1944')
+```
+
+The left of the box will be 6 June 1944 (D-Day) and for the right of the box we'll choose the first day of the Battle of the Bulge: 16 December 1944. In this case, the dates follow a month-day-year format, but `to_datetime` also works with [day-first and year-first formats](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.to_datetime.html). 
+
+We pass these coordinates to the `BoxAnnotation` constructor along with some styling arguments. Then, we add it to the our figure using the `add_layout()` method.
+
+```python
+box = BoxAnnotation(left=box_left, right=box_right,
+                    line_width=1, line_color='black', line_dash='dashed',
+                    fill_alpha=0.2, fill_color='orange')
+
+p.add_layout(box)
+```
+
+{% include figure.html filename="visualizing-with-bokeh-8.png" caption="A Time-Series Plot of the ETO with Annotations Added" %}
+
+<div class="alert alert-warning"> 
+
+Try to create a similar plot for the Pacific Theater of Operations (PTO). Annotate the invasion of Iwo Jima (February 19, 1945) and Japan's announcement of surrender (August 15, 1945).
+
+</div>
+
+# Spatial Data: Mapping Target Locations
+
+In this final part of the lesson we'll look at the spatial components of fragmentation bombs.
+
+Bokeh provides [built-in tile providers](https://bokeh.pydata.org/en/latest/docs/reference/tile_providers.html) that render base maps of the world. These are contained in the `bokeh.tile_providers` module. For this example, we'll use the CartoDB Tile Service (CARTODBPOSITRON).
+
+We'll also be using functions imported from the `pyproj` library. Since our coordinates are stored as latitude/longitude, we'll define a custom function to convert them before mapping. Note that although Bokeh is coordinate-system neutral, it uses the Web Mercator projection for mapping, a standard found across web tile providers. The subject of coordinate systems and projections are outside the scope of this tutorial, but the interested reader will find many useful web resources on these topics.
+
+<div class="alert alert-warning">
+
+If your own dataset has place names, but not latitude and longitude, don't worry! You can find ways to easily get coordinates from place names in Programming Historian's [Geocoding Historical Data using QGIS](programminghistorian.org/lessons/geocoding-qgis) or [Web Mapping with Python and Leaflet](https://programminghistorian.org/lessons/mapping-with-python-leaflet#geocoding-with-python).
```

Comment:
Looks like this line is causing the travis checks to fail. Here is the error - 

It looks this lesson contains a full link to "https://programminghistorian.org". All internal links should start with "/" followed by the relative page path, and not use the full domain name.

Reply:
Fixed. Thanks @walshbr.